### PR TITLE
Add release notes for 3.5.0

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

# Code review items

- Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- Documented: Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: release-notes/docs/3.5.0.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+- [Known issues](#known-issues)
+- [Upgrade notes](#upgrade-notes)
+- [Performance](#performance)
+- [Sort muted contacts](#sort-muted-contacts)
+- [SIH integration](#sih-integration)
+- [And more...](#and-more)
+
+## Known issues
+
+None.
```
Comment (kennsippell) [inline release-notes/docs/3.5.0.md]: might link the performance regression?

Comment (garethbowen) [reply release-notes/docs/3.5.0.md]: Good call. That came in after I wrote this...

File: release-notes/docs/3.5.0.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+| Metric | 3.4.0 | 3.5.0 | Improvement |
+|---|---|---|---|
+| Rendering the History tab | 7247ms | 1843ms | 75% |
+| Starting the application | 1020ms | 875ms | 14% |
+| Idle memory usage | 453MB | 202MB | 55% |
+
+A further optimization was made to the History detail pane by improving how we link to Patients, which reduced rendering time from 1002ms to 732ms an improvement of 27%. [medic#5455](https://github.com/medic/medic/issues/5455)
+
+### Sentinel processing
```
Comment (kennsippell) [inline release-notes/docs/3.5.0.md]: Maybe a more descript title here? Like `Optimizing Sentinel processing`?

File: release-notes/docs/3.5.0.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+There are no required changes to the [supported software matrix](https://github.com/medic/medic-docs/blob/master/installation/supported-software.md) from `3.0.0`.
+
+| Node | CouchDB | Browsers | SMS bridge | Android | medic-android | medic-couch2pg |
+|----|----|----|----|----|----|---|
+| 8.11+ | 2.1+ | Chrome 53+, Firefox latest | medic-gateway | 4.4+ | 0.4.5+ | 3.0+ |
+
+## Performance
+
+### Optimizing the History tab
```
Comment (kennsippell) [inline release-notes/docs/3.5.0.md]: Consistent casing in multi-word titles

File: release-notes/docs/3.5.0.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+| Starting the application | 1020ms | 875ms | 14% |
+| Idle memory usage | 453MB | 202MB | 55% |
+
+A further optimization was made to the History detail pane by improving how we link to Patients, which reduced rendering time from 1002ms to 732ms an improvement of 27%. [medic#5455](https://github.com/medic/medic/issues/5455)
+
+### Sentinel processing
+
+By reducing the the throttling of change processing and reducing the number of database writes we have greatly improved the rate at which Sentinel processes changes. Clearing a backlog of 200 documents used to take 67 seconds, but with 3.5.0 this has been reduced to 33 seconds, a 50% reduction. [medic#4128](https://github.com/medic/medic/issues/4128).
+
+To make Sentinel even more responsive to changes, transitions are now executed synchronously when messages are received. This means that replies are generated immediately and the user is be notified of the success or error of their message as soon as possible. [medic#4857](https://github.com/medic/medic/issues/4857).
```
Comment (kennsippell) [inline release-notes/docs/3.5.0.md]: typo `the user is be notified`
instead of `immediately` maybe `faster`

File: release-notes/docs/3.5.0.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+
+- [medic#1850](https://github.com/medic/medic/issues/1850): Report version of application and forms being used
+- [medic#4811](https://github.com/medic/medic/issues/4811): Expose SMS sending capabilities in android wrapper
+- [medic#4812](https://github.com/medic/medic/issues/4812): Support XForms that "compile" into SMS messages
+
+### Improvements
+
+- [horticulturalist#32](https://github.com/medic/horticulturalist/issues/32): Horti should by default install release version and not master
+- [medic-conf#117](https://github.com/medic/medic-conf/issues/117): Bulk user import
+- [medic-conf#135](https://github.com/medic/medic-conf/issues/135): Create-user function always creates a new place
```
Comment (kennsippell) [inline release-notes/docs/3.5.0.md]: Consider linking to https://github.com/medic/medic-conf/blob/master/release-notes.md#20

File: release-notes/docs/3.5.0.md (hunk 4 above)
Comment (garethbowen) [reply release-notes/docs/3.5.0.md]: I think it's correct to say "immediately". It's synchronous so immediately after the message is received we generate the response and save the doc. The bit that's merely faster is the notification being sent to the user...

File: release-notes/docs/3.5.0.md (hunk 5 above)
Comment (garethbowen) [reply release-notes/docs/3.5.0.md]: I'll have to think on this some more. My immediate thought is that medic-conf is a separate product so I don't want to confuse readers by linking there. But that means the issues should be included here either.

File: release-notes/docs/3.5.0.md (hunk 3 above)
Comment (garethbowen) [reply release-notes/docs/3.5.0.md]: Done.

File: release-notes/docs/3.5.0.md (hunk 2 above)
Comment (garethbowen) [reply release-notes/docs/3.5.0.md]: Done.

File: release-notes/docs/3.5.0.md (hunk 4 above)
Comment (garethbowen) [reply release-notes/docs/3.5.0.md]: Typo fixed though! Thanks.

**Pull request merged**